File: LinguaRise/linguarise.app/src/data/LessonSummaryDTO.js
import { WordDTO } from "../data/WordDTO";

export class LessonSummaryDTO {
constructor(data = {}) {
    this.languageName = data.languageName  ?? data.LanguageName  ?? "";
    this.flagImage    = data.flagImage     ?? data.FlagImage     ?? "";
    this.categoryName = data.categoryName  ?? data.CategoryName  ?? "";

    const raw = data.learnedWords ?? data.LearnedWords ?? [];
    this.learnedWords = Array.isArray(raw)
      ? raw.map(w => new WordDTO(w))
      : [];

    this.score = data.score ?? data.Score ?? 0;
  }
}

File: LinguaRise/LinguaRise.Models/DTOs/Lesson/LessonSummaryDTO.cs
namespace LinguaRise.Models.DTOs;

public class LessonSummaryDTO
{
    public string LanguageName { get; set; }
    public string? FlagImage { get; set; }
    public string CategoryName { get; set; }
    public ICollection<WordDTO> LearnedWords { get; set; } = new List<WordDTO>();
    public double Score { get; set; }
}
